src/praisonai/tests/unit/ui/test_ui_bind_aware_creds.py

```
@@ -0,0 +1,292 @@
+        # Should not raise
+        register_password_auth(None, bind_host="127.0.0.1")
+        
+        # Should have registered a callback
+        mock_cl.password_auth_callback.assert_called_once()
+    
+    @patch('praisonai.ui._auth.cl')
+    @patch.dict(os.environ, {}, clear=True)
+    def test_default_credentials_on_external_raises_error(self, mock_cl):
+        """Test registration with default credentials on external interface raises error."""
+        with pytest.raises(RuntimeError) as excinfo:
+            register_password_auth(None, bind_host="0.0.0.0")
```

Ruff (0.15.10)

[error] Possible binding to all interfaces

(S104)